### PR TITLE
feat: transform passes through upstream errors

### PR DESCRIPTION
**Pull request by Bluebie**
This was the most minimal version I could come up with which seems to get the job done. Messing with sink()'s implementation caused other tests to fail, and seeing as the issue is mainly going to come up with transform-it's anyway, it seems sensible to isolate the change to just `streamToIt.transform()`

Includes test and minimal change to make test pass.

### CODE REVIEW

**Comment by Bluebie** [comment]
Sorry if that got messy, turning on GitHub CI. Just wanted to verify everything was working across recent still supported versions of Node. Probably a good idea to turn it on on this repo too, so weird bugs in different node versions don't surprise anyone, and to make the status of CI easily accessible to everyone from the actions tab of the github repo page. It doesn't cost anything or have any downsides, for public repos.

**Comment by Bluebie** [comment]
Oh it's already running on travis? oops. I should pay more attention!

File: transform.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ module.exports = transform => async function * (source) {
   // In a transform the sink and source are connected, an error in the sink
   // will be thrown in the source also. Catch the sink error to avoid unhandled
   // rejections and yield from the source.
-  duplex.sink(source).catch(_ => {})
+  let sinkError
+  duplex.sink(source).catch(err => { sinkError = err })
+
   yield * duplex.source
```
Comment (alanshaw) [inline transform.js:9]: I'm just wondering if we should actually throw as soon as we can when we get an error instead of when the source has completed? i.e. expand this out with a for await and check on every iteration?

Comment (Bluebie) [reply transform.js:9]: I don't think it makes a big difference, because the error does destroy the Transform stream, causing it's asyncIterator to quickly become done, so usually the very next call to generator.next() will throw the error.

There maybe edge cases where destroy doesn't actually make the stream stop producing data. I could write some tests to verify that, but I doubt it'll be a big issue.

**Review by alanshaw — COMMENTED** [review]
This LGTM - thanks 🙏. Any chance you could remove the github actions config from the PR so we can get this merged? Either that or remove the travis config?

**Comment by Bluebie** [comment]
Github CI config removed

**Review by alanshaw — APPROVED** [review]
